src/ape/api/providers.py

```diff
@@ -1032,7 +1032,19 @@ def get_receipt(
         except TimeExhausted as err:
             raise TransactionNotFoundError(txn_hash) from err
 
-        txn = dict(self.web3.eth.get_transaction(HexStr(txn_hash)))
+        network_config = self.network.config.get(self.network.name)
+        max_retries = network_config.get("max_receipt_retries")
+        for attempt in range(max_retries):
+            try:
+                txn = dict(self.web3.eth.get_transaction(HexStr(txn_hash)))
+                break
+            except TransactionNotFound as e:
+                if attempt < max_retries - 1:  # if this wasn't the last attempt
+                    time.sleep(1)  # Wait for 1 second before retrying.
+                    continue  # Continue to the next iteration, effectively retrying the operation.
+                else:  # if it was the last attempt
+                    raise e  # Re-raise the last exception.
```

Comment:
```suggestion
                    raise  # Re-raise the last exception.
```

an empty raise is the same thing but preserves the exception stack!

Reply:
woah had no idea! thanks :)